src/assets/CollectiblesController.ts

```diff
@@ -1148,6 +1153,105 @@ export class CollectiblesController extends BaseController<
 
     this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
   }
+
+  /**
+   * Update colletible isTransacted status
+   *
+   * @param address - Hex address of the collectible contract.
+   * @param tokenId - Number that represents the id of the token.
+   * @param isTransacting - Collectible transacting status.
+   * @param transactionId - Hex id of the send transaction of the collectible.
+   */
+  updateCollectibleTransactionStatus(
+    address: string,
+    tokenId: string,
+    isTransacting: boolean,
+    transactionId: string,
+  ) {
+    const { allCollectibles } = this.state;
+    const { chainId, selectedAddress } = this.config;
+    const collectibles = allCollectibles[selectedAddress]?.[chainId] || [];
+    const index: number = collectibles.findIndex(
+      (collectible) =>
+        collectible.address === address && collectible.tokenId === tokenId,
+    );
+
+    if (index === -1) {
+      return;
+    }
+    const updatedCollectible: Collectible = {
+      ...collectibles[index],
+      isTransacting,
+      transactionId,
+    };
+
+    // Update Collectibles array
+    collectibles[index] = updatedCollectible;
+
+    this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
+  }
+  /**
+   * Reset the collectible transaction id value and turn false the isTransacting prop
+   *
+   * @param transactionId - Hex id of the send transaction of the collectible.
+   * @returns
+   */
+  resetCollectibleTransactionStatusByTransactionId(
+    transactionId: string,
```

Comment:
is this the best way of finding the collectible? Why not the `address` and `tokenId`?

Reply:
I created this one because, in the mobile app, where I needed to reset the transaction status, I only had the transaction Id data